### PR TITLE
Update Apache Lucene to 9.8.0-snapshot-95cdd2e

### PR DESCRIPTION
**Pull request by reta**
### Description
Update Apache Lucene to 9.8.0-snapshot-95cdd2e

### Related Issues
N/A

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [X] Commits are signed per the DCO using --signoff
- [ ] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: server/src/internalClusterTest/java/org/opensearch/action/admin/cluster/node/tasks/ConcurrentSearchTasksIT.java [hunk 1]
```diff
@@ -109,7 +109,7 @@ public void testConcurrentSearchTaskTracking() {
 
             Map<Long, List<ThreadResourceInfo>> threadStats = getThreadStats(SearchAction.NAME + "[*]", taskInfo.getTaskId());
             // Concurrent search forks each slice of 5 segments to different thread
-            assertEquals((int) Math.ceil(getSegmentCount(INDEX_NAME) / 5.0), threadStats.size());
+            assertEquals((int) Math.ceil(getSegmentCount(INDEX_NAME) / 5.0) + 1, threadStats.size());
```
Comment (reta) [inline server/src/internalClusterTest/java/org/opensearch/action/admin/cluster/node/tasks/ConcurrentSearchTasksIT.java:112]: @sohami as we know, in pre - Apache Lucene 9.8 the `Executor` was wrapped in `TaskExecutor` that invoked the last task on caller thread, this wrapping has been removed now.

**Comment by reta** [comment]
> * **RESULT:** FAILURE ❌

https://github.com/opensearch-project/OpenSearch/issues/10046

File: server/src/internalClusterTest/java/org/opensearch/action/admin/cluster/node/tasks/ConcurrentSearchTasksIT.java (hunk 1 above)
Comment (sohami) [reply server/src/internalClusterTest/java/org/opensearch/action/admin/cluster/node/tasks/ConcurrentSearchTasksIT.java:112]: lgtm. Can we also add reference to [this issue](https://github.com/apache/lucene/issues/12498) in the comment

**Comment by reta** [comment]
@msfroh mind please to quickly look through?

**Comment by reta** [comment]
@sohami mind please re-taking a look? thanks!

**Comment by reta** [comment]
@sohami @andrross could you help me to get it in folks? thank you!

**Review by kotwanikunal — APPROVED** [review]

**Pull request merged**